packages/cms/pages/resources/resource/resource.tsx

```diff
@@ -192,28 +193,17 @@ export const Table: React.FunctionComponent = () => {
   const [selectedItems, setSelectedItems] = useState<any[]>([])
   const [sortField, setSortField] = useState('firstName')
   const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>('asc')
+  const { resource, applyFilter } = useResourceStore()
 
   const columns: EuiBasicTableColumn<any>[] = [
-    {
-      field: 'firstName',
-      name: 'First name',
-      sortable: true,
-      truncateText: true
-    },
-    {
-      field: 'lastName',
-      name: 'Last name',
-      sortable: true,
-      truncateText: true
-    },
-    {
-      name: 'Date of birth',
-      field: 'dateOfBirth'
-    },
-    {
-      name: 'Nationality',
-      field: 'nationality'
-    },
+    ...(resource?.fields
+      .filter((resource, idx) => resource.showOnIndex && idx != 0)
```

Comment:
We are looping through `resource?.fields`, so a better variable name should be `field` and not `resource`. Also, even if it was supposed to be resource, we'll have to give it another name, so as to avoid shadow variables. since we have another variable called `resource`, it's not easily understandable to have another variable with same name. 


So here, I suggest we have `(resource?.fields.filter((field, idx) => field.showOnIndex)`

Reply:
Thanks, i will implement.